.github/workflows/gatk-tests.yml

```diff
@@ -0,0 +1,311 @@
+            bash --init-file /gatk/gatkenv.rc /root/run_unit_tests.sh;
+          TEST_EXIT_VALUE=$?;
+          $( exit ${TEST_EXIT_VALUE} );
+          sudo chmod -R a+w build/reports/;
+          mkdir build/reports/tests/test \
+           && cp -rp build/reports/tests/testOnPackagedReleaseJar/* build/reports/tests/test \
+           && rm -r build/reports/tests/testOnPackagedReleaseJar;
+
+      - uses: /broadinstitute/gatk/.github/actions/upload-results@lb_github_actions
+        if: always()
+        with:
+          warnPR: ${{ github.event_name == 'pull_request' && steps.docker-jacocco-tests.outcome != 'success' }}
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          jdk-version: ${{ matrix.Java }}
+          test-type: docker ${{ matrix.testType }}
+          job-matrix-id: ${{ github.run_id }}.${{ strategy.job-index }}
+          repo-path: ${{ github.ref_name }}_${{ github.run_id }}.${{ strategy.job-index }}
+          bot-comment-key: ${{ secrets.GATK_BOT_COMMENT_KEY }}
+          identifier: Java ${{ matrix.Java }} test on docker ${{matrix.testType}}
+
+
+  wdlGen:
+    runs-on: ubuntu-latest
+    name: Test WDL Generation
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch: 0
+      - name: Set up java 8
+        uses: actions/setup-java@v2
+        with:
+          java-version: '8'
+          distribution: 'adopt'
+          cache: gradle
+      - uses: /broadinstitute/gatk/.github/actions/install-cromwell@lb_github_actions
+        with:
+          CROMWELL_VERSION: ${{ env.CROMWELL_VERSION }}
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+
+      - name: Run WDL Generation Test
+        run: ./gradlew gatkWDLGenValidation
+
+  wdlTests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        wdlTest: [ 'RUN_CNV_GERMLINE_COHORT_WDL', 'RUN_CNV_GERMLINE_CASE_WDL', 'RUN_CNV_SOMATIC_WDL', 'RUN_M2_WDL', 'RUN_CNN_WDL' ]
+    continue-on-error: true
+    name: WDL test ${{ matrix.wdlTest }} on cromwell
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: pull lfs files
+        run: git lfs pull
+      - name: Set up java 8
+        uses: actions/setup-java@v2
+        with:
+          java-version: '8'
+          distribution: 'adopt'
+          cache: gradle
+      - uses: /broadinstitute/gatk/.github/actions/install-cromwell@lb_github_actions
+        with:
+          CROMWELL_VERSION: ${{ env.CROMWELL_VERSION }}
+
+      - name: "Exporting Environment Variables for Wdl Scripts"
+        run: |
+          if [[ "${{ github.event_name }}" == "pull_request" ]]
+          then
+            echo "CI_PULL_REQUEST=${{ github.event.number }}" >> $GITHUB_ENV 
+          fi
+          echo "CI_COMMIT=${{ github.sha }}" >> $GITHUB_ENV
+
+      - name: "Installing GATK for tests"
+        run: |
+          ./gradlew assemble;
+          ./gradlew installDist;
```

Comment:
This needs to assemble AND installDist AND shadowJar?  And then we ignore those and build a docker instead?  Do some wdl tests use the docker and others use the jar?  Can we move the build into just the parts that use it?

Reply:
This is a good point. I think we need assemble? this is ripped from the existing travis tests and this is exactly the set of steps we did before (also working in the constraints where most of the gradle tasks that involved the docs were failing when I made this which was causing crashes).

Reply:
So changing the wdl is a little bit of a pain and will involve a bunch more changes to that infrastructure. Currently they build the docker and then poke into the wdls the name of the docker with a defined pattern and i would have to see what can be done about that with the new repo.